ts/src/base/Exchange.ts

```diff
@@ -1922,6 +1925,22 @@ export default class Exchange {
         };
     }
 
+    async resetData () {
```

Comment:
We're missing two things here:
- cleaning ws structures
- using `createSafeDictionary` to the thread-safe dicts in c#

besides that I think we can call it just `clean` no?

Reply:
I think I've applied all changes you mentioned.
so, in py/php (where we have base inheritances) i've applied overrides for separate `close()` methods to close `WS/ASYNC/SYNC` instances, while for C#/JS we have only one `Close()` method that does centralized cleanup

Reply:
@carlosmiei I've pushed all updates and now it passed. whenever you can to revise this.

Reply:
ah wait, somewhat glitch, reverting into draft.